tiledb/sm/array_schema/array_schema.cc

```diff
@@ -592,9 +602,9 @@ Status ArraySchema::deserialize(ConstBuffer* buff) {
       return st_attr;
     }
 
-    Attribute* attr_ptr = tdb_new(Attribute, std::move(attr.value()));
-    attributes_.emplace_back(attr_ptr);
-    attribute_map_[attr_ptr->name()] = attr_ptr;
+    attributes_.emplace_back(
+        tdb::make_shared<Attribute>(HERE(), move(attr.value())));
+    attribute_map_[attributes_.back().get()->name()] = attributes_.back().get();
```

Comment:
I'm not convinced that an optimizer will detect that it only needs to compute `attributes_.back().get()` once. I'd use a temporary here, especially since as a pointer it's low-impact.